runtime/src/accounts_index.rs

```diff
@@ -1,38 +1,276 @@
 use solana_sdk::{clock::Slot, pubkey::Pubkey};
+use std::ops::{
+    Bound,
+    Bound::{Excluded, Included, Unbounded},
+};
 use std::sync::atomic::{AtomicU64, Ordering};
 use std::{
     collections::{BTreeMap, HashMap, HashSet},
     ops::RangeBounds,
-    sync::{RwLock, RwLockReadGuard},
+    sync::{Arc, RwLock},
 };
 
+const ITER_BATCH_SIZE: usize = 1000;
+
+rental! {
```

Comment:
interesting. :)

Reply:
thinking on this.... hmm.

Reply:
Updated with the ouroboros suggestion, thanks!